File: CS/ImageMemoEdit/CustomGridControl/CustomGridView/MyGridViewPrintInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Printing;
using EditorDescendant;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;

namespace ImageTextEdit
{
    public class MyGridViewPrintInfo : GridViewPrintInfo
    {
        public MyGridViewPrintInfo(PrintInfoArgs args)
            : base(args)
        {

        }
        public virtual new CustomGridView View { get { return (CustomGridView)base.View; } }
        GridCellInfo nowPrintingCellInfo;

        protected override void PrintRowCell(int rowHandle, GridCellInfo cell, System.Drawing.Rectangle r)
        {
            IIconSelector iconSelector = cell.Editor as IIconSelector;
            if (iconSelector != null)
            {
                MyRepositoryItemMemoEdit.IconSelectionEventHandler EH = new MyRepositoryItemMemoEdit.IconSelectionEventHandler(iconSelector_OnIconSelection);
                iconSelector.OnIconSelection += EH;
                nowPrintingCellInfo = cell;
                try
                {
                    base.PrintRowCell(rowHandle, cell, r);
                }
                finally
                {
                    iconSelector.OnIconSelection -= EH;
                }
            }
            else
                base.PrintRowCell(rowHandle, cell, r);

        }

        void iconSelector_OnIconSelection(object sender, IconSelectionEventArgs e)
        {
            SmartIconSelectionEventArgs ee = new SmartIconSelectionEventArgs(e.Image, nowPrintingCellInfo, false);
            View.RaiseOnSmartIconSelection(ee);
            e.Image = ee.Image;
        }
    }
}

File: VB/ImageMemoEdit/CustomGridControl/CustomGridView/MyGridViewPrintInfo.vb
Imports DevExpress.XtraGrid.Views.Printing
Imports EditorDescendant
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo

Namespace ImageTextEdit

    Public Class MyGridViewPrintInfo
        Inherits GridViewPrintInfo

        Public Sub New(ByVal args As PrintInfoArgs)
            MyBase.New(args)
        End Sub

        Public Overridable Overloads ReadOnly Property View As CustomGridView
            Get
                Return CType(MyBase.View, CustomGridView)
            End Get
        End Property

        Private nowPrintingCellInfo As GridCellInfo

        Protected Overrides Sub PrintRowCell(ByVal rowHandle As Integer, ByVal cell As GridCellInfo, ByVal r As System.Drawing.Rectangle)
            Dim iconSelector As IIconSelector = TryCast(cell.Editor, IIconSelector)
            If iconSelector IsNot Nothing Then
                Dim EH As MyRepositoryItemMemoEdit.IconSelectionEventHandler = New MyRepositoryItemMemoEdit.IconSelectionEventHandler(AddressOf iconSelector_OnIconSelection)
                AddHandler iconSelector.OnIconSelection, EH
                nowPrintingCellInfo = cell
                Try
                    MyBase.PrintRowCell(rowHandle, cell, r)
                Finally
                    RemoveHandler iconSelector.OnIconSelection, EH
                End Try
            Else
                MyBase.PrintRowCell(rowHandle, cell, r)
            End If
        End Sub

        Private Sub iconSelector_OnIconSelection(ByVal sender As Object, ByVal e As IconSelectionEventArgs)
            Dim ee As SmartIconSelectionEventArgs = New SmartIconSelectionEventArgs(e.Image, nowPrintingCellInfo, False)
            View.RaiseOnSmartIconSelection(ee)
            e.Image = ee.Image
        End Sub
    End Class
End Namespace
